src/ssz/tree_view/list_basic.zig

```diff
@@ -0,0 +1,217 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const hashing = @import("hashing");
+const Depth = hashing.Depth;
+const ListLengthUint = hashing.GindexUint;
+const Node = @import("persistent_merkle_tree").Node;
+const Gindex = @import("persistent_merkle_tree").Gindex;
+const isBasicType = @import("../type/type_kind.zig").isBasicType;
+
+const type_root = @import("../type/root.zig");
+const BYTES_PER_CHUNK = type_root.BYTES_PER_CHUNK;
+const itemsPerChunk = type_root.itemsPerChunk;
+const chunkDepth = type_root.chunkDepth;
+
+const BaseTreeView = @import("root.zig").BaseTreeView;
+const BasicPackedChunks = @import("chunks.zig").BasicPackedChunks;
+
+/// A specialized tree view for SSZ list types with basic element types.
+/// Elements are packed into chunks (multiple elements per leaf node).
+pub fn ListBasicTreeView(comptime ST: type) type {
+    comptime {
+        if (ST.kind != .list) {
+            @compileError("ListBasicTreeView can only be used with List types");
+        }
+        if (!@hasDecl(ST, "Element") or !isBasicType(ST.Element)) {
+            @compileError("ListBasicTreeView can only be used with List of basic element types");
+        }
+    }
+
+    return struct {
+        base_view: BaseTreeView,
+        list_length_cache: ?ListLengthUint = null,
+        prefetched_chunk_count: ?usize = null,
```

Comment:
I think we should continue with a single `TreeViewData` with no additional fields in the wrapping `TreeView` structs.
For example, I don't think we need either of these additional fields here or in the other `*TreeView` structs.

length is gindex 3, so it can use something like
```zig
pub fn length(self: *Self) !usize {
    const length_node = try self.base_view.getChildNode(@enumFromInt(3));
    const length_chunk = length_node.getRoot(self.base_view.pool);
    return std.mem.readInt(usize, length_chunk[0..@sizeOf(usize)], .little);
}
```

Likewise `prefetched_chunk_count` can be removed and a more naive implementation can be used.

We can bench and go further if needed (eg: using a union for `TreeViewData`), but lets only do that if we determine we need it.